mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxRecentDAOTest.java

```diff
@@ -94,14 +104,27 @@ public void removeFromRecentShouldOnlyRemoveUidWhenNotEmpty() throws Exception {
 
         testee.removeFromRecent(CASSANDRA_ID, UID2).join();
 
-        assertThat(testee.getRecentMessageUidsInMailbox(CASSANDRA_ID).join()).containsOnly(UID1);
+        assertThat(testee.getRecentMessageUidsInMailbox(CASSANDRA_ID).join()
+            .collect(Guavate.toImmutableList())).containsOnly(UID1);
     }
 
     @Test
     public void addToRecentShouldBeIdempotent() throws Exception {
         testee.addToRecent(CASSANDRA_ID, UID1).join();
         testee.addToRecent(CASSANDRA_ID, UID1).join();
 
-        assertThat(testee.getRecentMessageUidsInMailbox(CASSANDRA_ID).join()).containsOnly(UID1);
+        assertThat(testee.getRecentMessageUidsInMailbox(CASSANDRA_ID).join()
+            .collect(Guavate.toImmutableList())).containsOnly(UID1);
+    }
+
+    @Test
+    public void getRecentMessageUidsInMailboxShouldNotTimeoutWhenOverPagingLimit() throws Exception {
+        int size = 6000;
```

Comment:
I guess size must be superior of some constant on your code "paging limit". It would be great if you make that more explicit by refering to such a constant  when setting size